Source/Main.cpp

```diff
@@ -32,37 +32,57 @@ int main()
 }
 
 namespace {
+/// @brief Self declared function that loads in configuration files as needed.
+/// @param config 
 void loadConfig(Config &config)
 {
     std::ifstream configFile("config.txt");
     std::string key;
 
-    if (configFile.is_open()) {
-        while (configFile >> key) {
-            if (key == "renderdistance") {
-                configFile >> config.renderDistance;
-                std::cout << "Config: Render Distance: "
-                          << config.renderDistance << '\n';
-            }
-            else if (key == "fullscreen") {
-                configFile >> config.isFullscreen;
-                std::cout << "Config: Full screen mode: " << std::boolalpha
-                          << config.isFullscreen << '\n';
-            }
-            else if (key == "windowsize") {
-                configFile >> config.windowX >> config.windowY;
-                std::cout << "Config: Window Size: " << config.windowX << " x "
-                          << config.windowY << '\n';
-            }
-            else if (key == "fov") {
-                configFile >> config.fov;
-                std::cout << "Config: Field of Vision: " << config.fov << '\n';
+    try
+    {
+        if (configFile.is_open())
+        {
+            while (configFile >> key)
+            {
+                if (key == "renderdistance") {
+                    configFile >> config.renderDistance;
+                    std::cout << "Config: Render Distance: "
+                            << config.renderDistance << '\n';
+                }
+                else if (key == "fullscreen") {
+                    configFile >> config.isFullscreen;
+                    std::cout << "Config: Full screen mode: " << std::boolalpha
+                            << config.isFullscreen << '\n';
+                }
+                else if (key == "windowsize") {
+                    configFile >> config.windowX >> config.windowY;
+                    std::cout << "Config: Window Size: " << config.windowX << " x "
+                            << config.windowY << '\n';
+                }
+                else if (key == "fov") {
+                    configFile >> config.fov;
+                    std::cout << "Config: Field of Vision: " << config.fov << '\n';
+                }
             }
         }
+        else
+        {
+            std::cout << "Error: The program requires a configuration file to operate properly." << "\n";
+            std::cout << "Please find or clone the relevant config.txt file!" << "\n";
+            std::cout << "Terminating sequence..." << "\n";
+            throw "Unable to load configuration file. This process cannot continue.";
+        }
     }
-    else {
-        std::cerr << "Error: Could not find config.txt file! Using defaults.\n";
+    catch(const std::exception& e)
+    {
+        std::cerr << e.what();
     }
```

Comment:
For this, does it exit the program if the file is missing? I was thinking it could just create it using the default values if it is missing instead.

Reply:
That's not a bad idea. It wouldn't be hard to generate a pseudo configuration file before runtime if needed.

Reply:
I added the ability for Main to quickly write a replacement configuration file if one could not be used.

I hope this helps improve stability.